### PR TITLE
add twin list add tests and twin validation tests

### PR DESCRIPTION
**Pull request by Tschuck**
- test twin list add
- test twin validation
- use better xpath for label input selection
- add correct file-input id
- [CORE-848]

[CORE-848]: https://evannetwork.atlassian.net/browse/CORE-848

### CODE REVIEW

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 1]
```diff
@@ -1,22 +1,26 @@
-Feature: Display Digital Twin Details
+Feature: Digital Twin - Container + Datat Sets
 
   @tag:noLogout
+  @only
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
+    Then I want to see a text including "Full Example #3"
       And I want to see a text including "General"
-      And I want to see a text including "pluginLists"
       And I want to see a text including "pluginEntries"
+      And I want to see a text including "pluginEntriesValidation"
+      And I want to see a text including "pluginLists"
+      And I want to see a text including "pluginListsValidation"
 
   @tag:noLogout
+  @only
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 3]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
     Then The value of the Input field with label "textField" should be "gibberish"
     # object
     When I set Input field with label "objectField" to "gluposti"
-    Then I want to see a button "Save"
-      And I want to see a button "Cancel"
-      And I want to see a text including "Invalid JSON format."
+    Then I want to see a text including "Invalid JSON format."
     # And the button "Save" should be "disabled" #TODO
-    When I set Input field with label "objectField" to `{"new": "entry"}`
+    When I set Input field with label "objectField" to `{"new":"entry"}`
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]: no spaces allowed or just coincidence?

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 4]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
       And The value of the Input field with label "textField" should be "gibberish"
-      And The value of the Input field with label "objectField" should be `{"new": "entry"}`
+      And The value of the Input field with label "objectField" should be `{"new":"entry"}`
       And The value of the Input field with id "dataset-input-checkboxEntry-primitive" should be "true"
 
+  @tag:noLogout
+  Scenario: check formular validation
+    When I click on "pluginEntriesValidation" in side navigation
+      And I wait 30 seconds until loading was finished
+    When I click on input field with label "numberField"
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
      And I click on input field with label "numberField"
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 5]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+    Then the button "Save" should be "disabled"
+    # number
+    When I set Input field with label "numberField" to "12345"
+    Then I want to see a text including "should be <= 123"
+    When I set Input field with label "numberField" to "0"
+    Then I want to see a text including "should be >= 1"
+    And I set Input field with label "numberField" to "123"
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
    When I set Input field with label "numberField" to "123"
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 6]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+    # text
+    When I set Input field with label "textField" to "gibberish gibberish gibberish"
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
      And I set Input field with label "textField" to "gibberish gibberish gibberish"
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 7]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+    Then I want to see a text including "should not be longer than 20 characters"
+    When I set Input field with label "textField" to ""
+    Then I want to see a text including "should not be shorter than 1 character"
+      And I set Input field with label "textField" to "gibberish"
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
   When I set Input field with label "textField" to "gibberish"
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 8]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+    Then I want to see a text including "Invalid JSON format."
+    When I set Input field with label "objectField" to `{}`
+    Then I want to see a text including "should have required property bar"
+    When I set Input field with label "objectField" to `{"bar": "123"}`
+    Then I want to see a text including "should be number"
+    When I set Input field with label "objectField" to `{"bar": 123, "minMax": ""}`
+    Then I want to see a text including "should not be shorter than 1 character"
+    When I set Input field with label "objectField" to `{"bar": 123, "minMax": "12345678901"}`
+    Then I want to see a text including "should not be longer than 10 characters"
+      And I set Input field with label "objectField" to `{"bar": 123, "minMax": "123"}`
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
    When I set Input field with label "objectField" to `{"bar": 123, "minMax": "123"}`
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 9]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+    # array
+    When I set Input field with label "arrayField" to "gluposti"
+    Then I want to see a text including "Invalid JSON format."
+    When I set Input field with label "arrayField" to `[]`
+    Then I want to see a text including "should not have less than 1 item"
+    When I set Input field with label "arrayField" to `[""]`
+    Then I want to see a text including "should be object"
+    When I set Input field with label "arrayField" to `[{},{},{},{},{},{},{}]`
+    Then I want to see a text including "should not have more than 5 items"
+    And I set Input field with label "arrayField" to `[{}]`
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
    When I set Input field with label "arrayField" to `[{}]`
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 10]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+    # files
+    Then I want to see a text including "should not have less than 1 item"
+    When I upload file "bicycle.json" to the dropzone with the id "dataset-input-objectEntry-filesField"
+      And I wait for 1 seconds
+    When I click on button "Save"
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
      And I click on button "Save"
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 11]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+      And I wait 30 seconds until loading was finished
+
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]: no Then here?

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 12]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+  @tag:noLogout
+  @only
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
```

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature [hunk 13]
```diff
@@ -59,22 +63,89 @@ Feature: Display Digital Twin Details
...
+  Scenario: list all list plugin entries
+    When I click on "pluginListsValidation" in side navigation
+      And I want to see a text including "listObject"
```
Comment (karladler) [inline tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]:
```suggestion
   Then I want to see a text including "listObject"
```

File: tests-e2e/test-utils/test-utils.js [hunk 14]
```diff
@@ -134,17 +134,14 @@ export function getSelector(label, angular) {
...
     ];
 
-    return inputSelectors.map((inputSelector) => [
-      `//label[normalize-space(text()) = "${label}"]/preceding-sibling::${inputSelector}`,
-      `//label[normalize-space(text()) = "${label}"]/following-sibling::${inputSelector}`,
-      `//label/*[normalize-space(text()) = "${label}"]/parent::*/preceding-sibling::${inputSelector}`,
-      `//label/*[normalize-space(text()) = "${label}"]/parent::*/following-sibling::${inputSelector}`,
-    ].join('|')).join('|');
+    return inputSelectors
+      .map((inputSelector) => `//label[contains(., "${label}")]/parent::*//${inputSelector}`)
```
Comment (karladler) [inline tests-e2e/test-utils/test-utils.js]: cool if it works again :)

File: tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature (hunk 3 above)
Comment (Tschuck) [reply tests-e2e/features/assets/digital-twins/digital-twins-data-set.feature]: Was just for consistency.

**Pull request merged**